dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -737,6 +735,9 @@ public void delete(Context context, Collection collection) throws SQLException,
             owningCommunity.getCollections().remove(collection);
         }
 
+        // Remove all authorization policies
+        authorizeService.removeAllPolicies(context, collection);
+
```

Comment:
No longer needed. Policies should be cleaned up automatically now (see #1326 which added ON DELETE CASCADE for resource policies)